src/common/mfu_daos.c

```diff
@@ -1973,6 +1973,51 @@ static inline void init_hdf5_args(struct hdf5_args *hdf5)
     hdf5->ak = NULL;
 }
 
+static int serialize_kv_rec(struct hdf5_args *hdf5, 
+                            daos_key_t dkey,
+                            daos_handle_t *oh,
+                            uint64_t *dk_index,
+                            char *dkey_val,
+                            mfu_daos_stats_t* stats)
+{
+    void        *buf;
+    int         rc;
+    herr_t      err;
+    hvl_t       *kv_val;
+    daos_size_t size = 0;
+
+    /* get the size of the value */
+    rc = daos_kv_get(*oh, DAOS_TX_NONE, 0, dkey_val, &size, buf, NULL);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to fetch object");
+        goto out;
+    }
+    buf = MFU_CALLOC(1, size);
+    if (buf == NULL) {
+        rc = ENOMEM;
```

Comment:
@dsikich Since the daos_ functions return -DER codes, we should probably do the same:
```suggestion
        rc = -DER_NOMEM;
```